fishgen/src/lib.rs

```diff
@@ -99,3 +102,147 @@ pub fn random_fish(
+    })
+}
+
+pub fn random_squiggle(
+    width: u32,
+    height: u32,
+    min_scale: f32,
+    max_scale: f32,
+) -> Result<FishOutput, FishError> {
+    if (width as f32) < FISH_WIDTH as f32 * max_scale ||
+       (height as f32) < FISH_HEIGHT as f32 * max_scale
+    {
+        return Err(FishError::FishTooBig {
+            max_supported_scale: f32::min(
+                width as f32 / FISH_WIDTH as f32,
+                height as f32 / FISH_HEIGHT as f32,
+            ),
+        });
+    }
+
+    let scale = fastrand::f32() * (max_scale - min_scale) + min_scale;
+
+    let fish_width = (FISH_WIDTH as f32 * scale) as u32;
+    let fish_height = (FISH_HEIGHT as f32 * scale) as u32;
+
+    let fish = rotate_about_center( // note: this is not a fish
+        &(SQUIGGLE.clone().to_rgba8()),
+        fastrand::f32() * std::f32::consts::TAU,
+        Interpolation::Bicubic,
+        Rgba::<u8>([255, 255, 255, 0]),
+    );
+
+    let fish = resize(&fish, fish_width, fish_height, FilterType::Triangle);
+
+    let mut image: ImageBuffer<Rgba<u8>, _> =
+        image::ImageBuffer::from_pixel(width, height, Rgba::<u8>([255, 255, 255, 255]));
+
+    let fish_x = fastrand::u32(0..(width - fish_width));
+    let fish_y = fastrand::u32(0..(width - fish_width));
+
+    overlay(&mut image, &fish, fish_x.into(), fish_y.into());
+
+    Ok(FishOutput {
+        image: image.convert(),
+        x: fish_x,
+        y: fish_y,
+        width: fish_width,
+        height: fish_height,
+    })
+}
+
+pub fn random_decoy_fish(
+    width: u32,
+    height: u32,
+    min_scale: f32,
+    max_scale: f32,
+) -> Result<FishOutput, FishError> {
+    if (width as f32) < FISH_WIDTH as f32 * max_scale ||
+       (height as f32) < FISH_HEIGHT as f32 * max_scale
+    {
+        return Err(FishError::FishTooBig {
+            max_supported_scale: f32::min(
+                width as f32 / FISH_WIDTH as f32,
+                height as f32 / FISH_HEIGHT as f32,
+            ),
+        });
+    }
+
+    let scale = fastrand::f32() * (max_scale - min_scale) + min_scale;
+
+    let fish_width = (FISH_WIDTH as f32 * scale) as u32;
+    let fish_height = (FISH_HEIGHT as f32 * scale) as u32;
+
+    let fish = rotate_about_center( // note: this is not a fish
+        &(DECOY.clone().to_rgba8()),
+        fastrand::f32() * std::f32::consts::TAU,
+        Interpolation::Bicubic,
+        Rgba::<u8>([255, 255, 255, 0]),
+    );
+
+    let fish = resize(&fish, fish_width, fish_height, FilterType::Triangle);
+
+    let mut image: ImageBuffer<Rgba<u8>, _> =
+        image::ImageBuffer::from_pixel(width, height, Rgba::<u8>([255, 255, 255, 255]));
+
+    let fish_x = fastrand::u32(0..(width - fish_width));
+    let fish_y = fastrand::u32(0..(width - fish_width));
+
+    overlay(&mut image, &fish, fish_x.into(), fish_y.into());
+
+    Ok(FishOutput {
+        image: image.convert(),
+        x: fish_x,
+        y: fish_y,
+        width: fish_width,
+        height: fish_height,
+    })
+}
```

Comment:
Could the core put-image-on-white-background be separated, and have these extra functions (+ the fish one) be wrappers over that?

Reply:
this is indeed a good idea, will do